rust/lance-io/src/object_store.rs

```diff
@@ -103,6 +103,21 @@ impl std::fmt::Display for ObjectStore {
     }
 }
 
+pub trait ObjectStoreProvider: std::fmt::Debug + Sync + Send {
+    fn new_store(&self, base_path: Url) -> Result<ObjectStore>;
```

Comment:
At a minimum, I think `new_store()` should also take `storage_options: HashMap<String, String>`. Though probably better to take the full `&ObjectStoreParams`. Ideally, we should be able to `configure_store` to use the registry for all of the schemes.

Reply:
done 👍